packages/web/src/pages/collection-page/CollectionPageProvider.tsx

```diff
@@ -1012,8 +1016,19 @@ function mapDispatchToProps(dispatch: Dispatch) {
       ),
     updatePlaylistLastViewedAt: (playlistId: ID) =>
       dispatch(updatedPlaylistViewed({ playlistId })),
-    openPremiumContentPurchaseModal: (args: PremiumContentPurchaseModalState) =>
+    openPremiumContentPurchaseModal: (
+      args: PremiumContentPurchaseModalState & { source: ModalSource }
+    ) => {
+      // Since we cant use the premium modal hook we have to manually trigger the modal action & the analytics tracking call
```

Comment:
don't quite understand this, don't we have modal analytics baked in already? are you doing this specifically to get `source` to be tracked?

Reply:
its baked into the hook, not in the action itself. Since this is a class component we don't get the baked in analytics 😞